### PR TITLE
Add PEP 613 TypeAlias to typing_extensions

### PR DESCRIPTION
**Pull request by east825**

### CODE REVIEW

File: typing_extensions/src_py2/test_typing_extensions.py [hunk 1]
```diff
@@ -377,6 +377,44 @@ def test_annotated_in_other_types(self):
         self.assertEqual(X[int], List[Annotated[int, 5]])
 
 
+class TypeAliasTests(BaseTestCase):
```
Comment (gvanrossum) [inline typing_extensions/src_py2/test_typing_extensions.py:380]: Could you add some positive tests? I've forgotten the syntax we agreed upon and there's no hint in this PR of what it should be.

File: typing_extensions/src_py2/typing_extensions.py [hunk 2]
```diff
@@ -241,5 +241,43 @@ class Annotated(object):
...
+class _TypeAliasBase(typing._FinalTypingBase):
+    """Special marker indicating that an assignment should
+    be recognized as a proper type alias definition by type
+    checkers.
+
+    For example::
+
+        Predicate = Callable[[...], bool]  # type: TypeAlias
+
+    It's invalid when used anyhow but as in the example above.
```
Comment (gvanrossum) [inline typing_extensions/src_py2/typing_extensions.py:265]: anyhow but -> anywhere except

File: typing_extensions/src_py3/typing_extensions.py [hunk 3]
```diff
@@ -2071,3 +2071,100 @@ def get_args(tp):
...
+    def TypeAlias(self, parameters):
+        """Special marker indicating that an assignment should
+        be recognized as a proper type alias definition by type
+        checkers.
+
+        For example::
+
+            Predicate: TypeAlias = Callable[[...], bool]
+
+        It's invalid when used anyhow but as in the example above.
```
Comment (gvanrossum) [inline typing_extensions/src_py3/typing_extensions.py:2093]: anyhow as -> anywhere except

File: typing_extensions/src_py3/typing_extensions.py [hunk 4]
```diff
@@ -2071,3 +2071,100 @@ def get_args(tp):
...
+    class _TypeAliasBase(typing._FinalTypingBase, metaclass=_TypeAliasMeta, _root=True):
+        """Special marker indicating that an assignment should
+        be recognized as a proper type alias definition by type
+        checkers.
+
+        For example::
+
+            Predicate: TypeAlias = Callable[[...], bool]
+
+        It's invalid when used anyhow but as in the example above.
```
Comment (gvanrossum) [inline typing_extensions/src_py3/typing_extensions.py:2131]: (again)

File: typing_extensions/src_py3/typing_extensions.py [hunk 5]
```diff
@@ -2071,3 +2071,100 @@ def get_args(tp):
...
+    class TypeAlias(metaclass=_TypeAliasMeta, _root=True):
+        """Special marker indicating that an assignment should
+        be recognized as a proper type alias definition by type
+        checkers.
+
+        For example::
+
+            Predicate: TypeAlias = Callable[[...], bool]
+
+        It's invalid when used anyhow but as in the example above.
```
Comment (gvanrossum) [inline typing_extensions/src_py3/typing_extensions.py:2168]: (ditto)

**Comment by east825** [comment]
Oh, sorry, I missed the notifications. 

I've added positive tests and fixed docstrings as suggested. Also, there happened to be a mistake in `Callable` syntax (shame on me) and a few code style inconsistencies that I overlooked initially. 

I pushed everything as individual commits to make the changes more evident as e.g. typeshed contributing guidelines require, let me know if I need to squash them manually.

**Comment by gvanrossum** [comment]
@east825 Please don't squash, we do that when we merge. I am behind on code reviews, ping me later this week.

**Pull request merged**